### PR TITLE
BIP 310: Stratum protocol extensions

### PR DESCRIPTION
**Pull request by honzik666**
Hi,

This proposal went formally through the mailing list: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-May/015978.html without any further discussion. Currently, this extension has been implemented by multiple pools. I would like to kindly ask for BIP # assignment and have this merged into upstream bips.

Best regards,

Jan Capek

### CODE REVIEW

**Comment by achow101** [comment]
As an aside, perhaps the stratum protocol itself should be specified in BIP documents as well so that people don't have to go to other sites to learn about the stratum protocol itself before reading the extensions BIP. The Stratum protocol has BIPs 40 and 41 but there are no documents for those present in the repo.

**Comment by honzik666** [comment]
That is a good point, we can compile a stratum BIP from the spec that we published here: http://stratumprotocol.org/

**Comment by luke-jr** [comment]
Please use BIP number 310.

Also, while this does address backward compatibility to a limited degree, it should ideally be discussed in a dedicated section of the document. Ideally, even if you insist on breaking compatibility with the established (and technically superior) `mining.capabilities` for some reason, it would be nice to document that incompatibility there as well.

**Comment by honzik666** [comment]
On Thu, 09 Aug 2018 11:11:22 -0700
Luke Dashjr <notifications@github.com> wrote:

> Please use BIP number 310.
> 
> Also, while this does address backward compatibility to a limited
> degree, it should ideally be discussed in a dedicated section of the
> document. Ideally, even if you insist on breaking compatibility with
> the established (and technically superior) `mining.capabilities` for
> some reason, it would be nice to document that incompatibility there
> as well.
> 
Thank you for assigning it, I will try to document this, that is a good
point.


-- 
CEO Braiins Systems | Slushpool.com
email: jan.capek@braiins.cz
http://braiins.cz
http://slushpool.com

File: bip-0310.mediawiki [hunk 1]
```diff
@@ -271,6 +271,15 @@ Miner provides additional text-based information.
 * '''"info.hw-id"''' (OPTIONAL, ''String'')
 ::- Unique  identifier of the mining device
 
+==Compatibility==
+
+Currently, there is a similar protocol feature '''mining.capabilities''' that
+was intended for various protocol extensions.  However, '''mining.configure'''
+is incompatible with this feature as it requires a server response confirming
+all accepted/negotatied extensions. The reason why we made it incompatible is
+that '''mining.capabilities''' request has no associated response.
```
Comment (luke-jr) [inline bip-0310.mediawiki]: It has no defined response, but it would have been trivial to add one...

File: bip-0310.mediawiki [hunk 2]
```diff
@@ -0,0 +1,285 @@
+<pre>
+  BIP: 0310
```
Comment (luke-jr) [inline bip-0310.mediawiki]: Shouldn't have a leading zero here

**Pull request merged**